firmware/Core/Src/sys_reboot_reason.c

```diff
@@ -0,0 +1,113 @@
+// Source: https://stackoverflow.com/questions/34196663/
+
+#include "sys_reboot_reason.h"
+
+#include "main.h"
+
+/// @brief      Obtain the STM32 system reset cause
+/// @param      None
+/// @return     The system reset cause
+reset_cause_t reset_cause_get(void)
+{
+	reset_cause_t reset_cause;
+
+	if (__HAL_RCC_GET_FLAG(RCC_FLAG_LPWRRST))
+	{
+		reset_cause = RESET_CAUSE_LOW_POWER_RESET;
+	}
+	else if (__HAL_RCC_GET_FLAG(RCC_FLAG_WWDGRST))
+	{
+		reset_cause = RESET_CAUSE_WINDOW_WATCHDOG_RESET;
+	}
+	else if (__HAL_RCC_GET_FLAG(RCC_FLAG_IWDGRST))
+	{
+		reset_cause = RESET_CAUSE_INDEPENDENT_WATCHDOG_RESET;
+	}
+	else if (__HAL_RCC_GET_FLAG(RCC_FLAG_SFTRST))
+	{
+		// This reset is induced by calling the ARM CMSIS 
+		// `NVIC_SystemReset()` function!
+		reset_cause = RESET_CAUSE_SOFTWARE_RESET; 
+	}
+	else if (__HAL_RCC_GET_FLAG(RCC_FLAG_BORRST))
+	{
+		reset_cause = RESET_CAUSE_BROWNOUT_RESET; // POR/PDR or BOR reset
+	}
+	else if (__HAL_RCC_GET_FLAG(RCC_FLAG_PINRST))
+	{
+		
+		reset_cause = RESET_CAUSE_EXTERNAL_RESET_PIN_RESET;
+	}
+    else if (__HAL_RCC_GET_FLAG(RCC_FLAG_OBLRST))
+    {
+        reset_cause = RESET_CAUSE_OPTION_BYTE_LOADER_RESET;
+    }
+    else if (__HAL_RCC_GET_FLAG(RCC_FLAG_FWRST)){
+        reset_cause = RESET_CAUSE_FIREWALL_RESET;
+    }
+	else
+	{
+		reset_cause = RESET_CAUSE_UNKNOWN;
+	}
+
+	return reset_cause; 
+}
+
+// Note: any of the STM32 Hardware Abstraction Layer (HAL) Reset and Clock
```

Comment:
Move comment to relevant section.

Is this comment relevant? It's referencing the `STM32F7` MCU; we have an STM32L4.